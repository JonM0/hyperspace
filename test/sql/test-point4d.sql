select '(0,1,1,0)'::point4d;

select p.p[0], p.p[1], p.p[2], p.p[3]
from (select '(1,2,3,4)'::point4d as p);

select ('(1,2,3,4)'::point4d)[-1];
select ('(1,2,3,4)'::point4d)[200000];

create table t (
    a point4d
);

insert into t values
    ('(0,0,0,1)'),
    ('(0,0,2,2)'),
    ('(1,1,0.5,3)'),
    ('(3,-9,2,32)');

select sum(a) from t;

drop table t;