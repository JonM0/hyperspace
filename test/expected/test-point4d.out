select '(0,1,1,0)'::point4d;
  point4d  
-----------
 (0,1,1,0)
(1 row)

select p.p[0], p.p[1], p.p[2], p.p[3]
from (select '(1,2,3,4)'::point4d as p) as p;
 p | p | p | p 
---+---+---+---
 1 | 2 | 3 | 4
(1 row)

select ('(1,2,3,4)'::point4d)[-1];
 point4d 
---------
        
(1 row)

select ('(1,2,3,4)'::point4d)[200000];
 point4d 
---------
        
(1 row)

create table t (
    a point4d
);
insert into t values
    ('(0,0,0,1)'),
    ('(0,0,2,2)'),
    ('(1,1,0.5,3)'),
    ('(3,-9,2,32)');
select sum(a) from t;
      sum      
---------------
 (4,-8,4.5,38)
(1 row)

drop table t;
